compiler/ir/ir.psi2ir/src/org/jetbrains/kotlin/psi2ir/generators/PropertyGenerator.kt

```diff
@@ -171,5 +173,11 @@ class PropertyGenerator(declarationGenerator: DeclarationGenerator) : Declaratio
         val variableDescriptor = getOrFail(BindingContext.VARIABLE, ktProperty)
         return variableDescriptor as? PropertyDescriptor ?: TODO("not a property: $variableDescriptor")
     }
+
+    private val Visibility.admitsFakeOverride: Boolean
+        get() = this != Visibilities.PRIVATE && this != Visibilities.PRIVATE_TO_THIS && this != Visibilities.INVISIBLE_FAKE
```

Comment:
Looks like this can be simplified to `!Visibilities.isPrivate(this)` and inlined because invisible fakes have been filtered out in the first line of `generateFakeOverrideProperty`?

Reply:
That's what I wrote originally! However, if you have a lateinit field with a private setter then the field visibility in a subclass is "invisible fake". This is because we take the visibility of the setter as the visibility of the field.

So there's either going to be a special case in `computeFieldVisibility` or we add a utility function which sees invisible fakes as private. I would prefer the latter.

Reply:
Thanks! I agree. I'd still prefer replacing two checks with `Visibilities.isPrivate`, but I don't insist

Reply:
Done